lib/pavilion/plugins/commands/run.py

```diff
@@ -18,6 +19,18 @@
 from pavilion.test_run import TestRun, TestRunError
 
 
+def build_test_thread(test, failed_list, err_out):
+
+    if not test.build():
+        fprint("Error building test: ", file=err_out, color=output.RED)
```

Comment:
You should surround all the output with a lock, to make sure it doesn't get interleaved.